packages/framer-motion/src/components/AnimatePresence/index.tsx

```diff
@@ -185,19 +185,37 @@ export const AnimatePresence: React.FunctionComponent<
         let exitingComponent = component
         if (!exitingComponent) {
             const onExit = () => {
-                allChildren.delete(key)
+                // clean up the exiting children map
                 exitingChildren.delete(key)
 
-                // Remove this child from the present children
-                const removeIndex = presentChildren.current.findIndex(
-                    (presentChild) => presentChild.key === key
+                // compute the keys of children that were rendered once but are no longer present
+                // this could happen in case of too many fast consequent renderings
+                // @link https://github.com/framer/motion/issues/2023
+                const leftOverKeys = Array.from(allChildren.keys()).filter(
+                    (childKey) => !targetKeys.includes(childKey)
+                )
+
+                // clean up the all children map
+                leftOverKeys.forEach((leftOverKey) =>
+                    allChildren.delete(leftOverKey)
```

Comment:
Note that the `allChildren` map was never cleaned up correctly and it was kept growing during all the rendering cycles. This patch fixes this memory leak as well